File: src/dotnet/Taml.NET/Taml.NET/TamlKeyValuePair.cs
namespace TAML
{
	public class TamlKeyValuePair : TamlValue
	{
		public TamlKeyValuePair(string key, TamlValue? value)
		{
			this.Key = key;
			this.Value = value;
		}

		public string Key { get; set; }

		public TamlValue? Value { get; set; }

		public override string ToString()
		{
			return $"{Key}: {Value?.ToString()}";
		}

		public static implicit operator string(TamlKeyValuePair value)
		{
			return value.ToString();
		}


	}


}

File: src/JVM/taml4j-parser/src/main/java/com/taml4j/TamlKeyValuePair.java
package com.taml4j;

/**
 * @author Luca Taddeo - Lucalas
 */
public class TamlKeyValuePair extends TamlValue implements KeyValuePairInterface<String, TamlValue>  {

	protected String key;
	protected TamlValue value;

	public TamlKeyValuePair(String key, TamlValue value) {
		this.key = key;
		this.value = value;
	}

	public void setKey(String key) {
		this.key = key;
	}

	public String getKey() {
		return key;
	}

	public TamlValue getValue() {
		return value;
	}

	public TamlValue setValue(TamlValue value) {
		this.value = value;
		return value;
	}

	public boolean hasValue() {
		return value != null;
	}

	@Override
	public String toString() {
		return hasValue() ? key + ": " + value : key;
	}

}
